### DISCUSSION TITLE
gramps/topola: where is the file stored that the local webserver uses?

### DISCUSSION THREAD

**Discussion by allangould** [PeWu/topola-viewer#136] [Q&A] [answered]
I am using Topola Viewer from gramps (on Windows; GRAMPS: AIO64-5.1.4-1 ) as "Interactive family tree".

gramps “Interactive family tree” creates a gedcom file, does a bit of post processing on the gedom file, then starts a web server, then uses topoloa-viewer to view the gedcom file. So I'm guessing that there's a temporary gedcom file stored somewhere on the local machine? Where is the (temporary) gedcom file stored (within the gramps environment/on the local machine/filestore - as it’s clearly stated that no data is exported?), so it must presumably be somewhere local for the local webserver. TIA

**Comment by czifumasa** [comment, marked as answer]
In gramps addon for Topola, Gedcom file is created using python function [mkstemp](https://docs.python.org/3/library/tempfile.html#tempfile.mkstemp). This function creates temporary file in default temp directory according to your OS. In my environment (Windows 10), file is created in `%userprofile%\AppData\Local\Temp`. The name of the file is random generated hash with 12 characters and file doesn't have an extension, but when you open it in notepad it looks like regular gedcom file.
